x/incentives/keeper/iterator.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/c-osmosis/osmosis/x/incentives/types"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) iteratorAfterTime(ctx sdk.Context, prefix []byte, time time.Time) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	timeKey := getTimeKey(time)
+	key := combineKeys(prefix, timeKey)
+	return store.Iterator(storetypes.InclusiveEndBytes(key), storetypes.PrefixEndBytes(prefix))
+}
+
+func (k Keeper) iteratorBeforeTime(ctx sdk.Context, prefix []byte, time time.Time) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	timeKey := getTimeKey(time)
+	key := combineKeys(prefix, timeKey)
+	return store.Iterator(prefix, storetypes.InclusiveEndBytes(key))
+}
+
+func (k Keeper) iterator(ctx sdk.Context, prefix []byte) sdk.Iterator {
+	store := ctx.KVStore(k.storeKey)
+	return sdk.KVStorePrefixIterator(store, prefix)
+}
+
+// UpcomingPotsIteratorAfterTime returns the iterator to get upcoming pots that start distribution after specific time
+func (k Keeper) UpcomingPotsIteratorAfterTime(ctx sdk.Context, time time.Time) sdk.Iterator {
+	return k.iteratorAfterTime(ctx, types.KeyPrefixUncomingPots, time)
+}
+
+// UpcomingPotsIteratorBeforeTime returns the iterator to get upcoming pots that already started distribution before specific time
+func (k Keeper) UpcomingPotsIteratorBeforeTime(ctx sdk.Context, time time.Time) sdk.Iterator {
+	return k.iteratorBeforeTime(ctx, types.KeyPrefixUncomingPots, time)
+}
+
+// UpcomingPotsIterator returns iterator for upcoming pots
+func (k Keeper) UpcomingPotsIterator(ctx sdk.Context) sdk.Iterator {
```

Comment:
I think its fine if we remove the `Iterator` suffix from all of these functions, what do you think?

Reply:
It's used in the Cosmos SDK, example: https://github.com/cosmos/cosmos-sdk/blob/master/x/gov/keeper/keeper.go#L153

Reply:
Hrmm, ok seems fine to leave it for now. I'm not generally convinced about this being our public API for the module, but I don't think its critical for now.